js/tools/eraser.js

```diff
@@ -0,0 +1,33 @@
+/* global app */
+window.tools.eraser = {
+  name: 'eraser',
+  icon: '/images/eraser.png',
+  state: {
+    selected: false,
+    mousePressed: false
+  },
+  events: {
+    mousedown: function (e, self) {
+      self.state.mousePressed = true
+    },
+    mouseup: function (e, self) {
+      self.state.mousePressed = false
+    },
+    mousemove: function (e, self) {
+    // if self tool is selected AND the mouse is pressed
+      if (self.state.selected && self.state.mousePressed) {
+        const mouse = app.eventToMouse(e)
+        const radius = app.ctx.lineWidth * 10
+        const oldFill = app.ctx.fillStyle
+        app.ctx.globalAlpha = 0.5
+        app.ctx.fillStyle = 'white'
+        app.ctx.beginPath()
+        app.ctx.arc(mouse.x, mouse.y, radius, 0, 2 * Math.PI, false)
+        app.ctx.fill()
+        // app.ctx.lineWidth = 5
+        app.ctx.fillStyle = oldFill
+        // app.ctx.globalAlpha = 1
+      }
```

Comment:
this works great, but it has a side-effect. b/c u're changing the `app.ctx.globalAlpha = 0.5` in ur brush it will unintentionally effect everyone else's brushes as well, i can see that u're trying to account for this w/ur `oldFill` variable (good thinking!), but it's not quite working as intended. in order to solve this u can use the Canvas API's [save / restore](https://www.tutorialspoint.com/html5/canvas_states.htm) state methods.

inside ur mousemove function's conditional statement place a call to `app.ctx.save()` somewhere before u change the global alpha, then after u run ur fill function place a call to `app.ctx.restore()`